src/gsi/update_guess.f90

```diff
@@ -281,9 +289,39 @@ subroutine update_guess(sval,sbias)
            endif
            icloud=getindex(cloud,guess(ic))
            if(icloud>0) then
-                 ptr3dges = max(ptr3dges+ptr3dinc,zero)
-              cycle
-           else  
+              if (trim(guess(ic))=='qr' .or. &
```

Comment:
This part of code can be simplified by checking the "l_use_dbz_directDA" first. All Original part can be combined as its original form. In "l_use_dbz_directDA" block, qr,qs,qg will be a part and others can go to one else block.

Reply:
I added explanation of parameters in comment block according to your comments.

https://github.com/NOAA-EMC/GSI/pull/142/files#diff-fa08cb92718d118dfe569a760912124e0d84ebdddad08003381497e6c4407a92R1036-R1232